src/Call.tsx

```diff
@@ -495,10 +493,14 @@ export class Call extends EventEmitter {
 
     this._update(nativeCallEvent);
 
-    const error = new GenericError(
-      nativeCallEvent.error.message,
-      nativeCallEvent.error.code
-    );
+    const { message, code } = nativeCallEvent.error;
+
+    const ErrorClass = errorsByCode.get(code);
+
+    const error = ErrorClass
+      ? new ErrorClass(message)
+      : new TwilioError(message, code);
```

Comment:
This is duplicated multiple times in Call.tsx and Voice.tsx. Continuing to add new code that emits error objects meaning duplicating this more. Can you add a utility function or similar that does this? Something like
```
const error = createTwilioError(nativeErrorObject);
```